src/server/main.py

```diff
@@ -30,13 +31,14 @@ def _get_taxon(params, headers):
         'type': 'object',
         'required': ['id'],
         'properties': {
-            'id': {'type': 'string', 'pattern': id_pattern}
+            'id': {'type': 'string', 'pattern': id_pattern},
+            'ts': {'type': 'integer', 'minimum': 0},
         }
     }
     jsonschema.validate(instance=params, schema=schema)
-    (coll, key) = params['id'].split('/')
-    params['key'] = key
-    del params['id']
+    (coll, _id) = params['id'].split('/')
```

Comment:
This code is repeated over and over, personally I'd make a method

Reply:
Made it into a function